datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/concurrency/GraphQLWorkerPoolThreadFactory.java

```diff
@@ -0,0 +1,25 @@
+package com.linkedin.datahub.graphql.concurrency;
+
+import java.util.concurrent.ThreadFactory;
+
+public class GraphQLWorkerPoolThreadFactory implements ThreadFactory {
+
+  private static int threadInitNumber;
+
+  private static synchronized int nextThreadNum() {
+    return threadInitNumber++;
```

Comment:
Use [AtomicLong](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicLong.html) instead and return `threadInitNumber.getAndIncrement()`?

Reply:
Sure, I just copied what the base ThreadFactory impl did here.